### DISCUSSION TITLE
Using automation output in UI

### DISCUSSION THREAD

**Discussion by avivkri** [Budibase/budibase#6988] [Help] [answered]
Is it possible to use the output of an automation action in UI without saving to database (just like query outputs)?

I do not see any bindings suggested automatically for the previous automation action.
![image](https://user-images.githubusercontent.com/13772138/181786206-d2425081-3aff-402d-8a1a-98e275279c2d.png)

**Comment by melohagan** [comment, marked as answer]
Hey @avivkri 

No it's not possible to get a callback from an automation to the UI. This is because automations are run in the background, independently from the UI.

Possibly in the future we could introduce pub/sub or websocket functionality, but that's not available at the moment.

**Reply by avivkri** [reply]
Hi @melohagan,

Thanks for the quick support. That clarifies my situation.

I went with the automation in the first place because REST form-data does not allow attaching files at the moment. So, I turned to automation with a bash script to cURL the request manually. And that's where I needed to use the automation response in UI.

Could you suggest anything for this situation?

**Reply by melohagan** [reply]
There is an attachment component: https://docs.budibase.com/docs/attachment

You may be able to use an **On Change** event on that field to grab some meta data upon upload if that's all you need?
https://docs.budibase.com/docs/actions#on-change

**Reply by avivkri** [reply]
@melohagan,

I'm afraid I do not know if the quoted issue #6975 is related here.

I can explain further to avoid confusion. I meant the multipart form-data body in REST query allows only string values today, neither files nor attachment components.
![POST form-data with file](https://user-images.githubusercontent.com/13772138/181799992-359f6cc0-0ef0-40e4-b07c-349b89496b21.jpg)

**Reply by melohagan** [reply]
Apologies, I meant to link the docs page. I've updated my comment.

**Reply by avivkri** [reply]
No problem.

I've already used the attachment component that you suggest.

My question is:
In the screenshot below, the **multipart form-data body in REST query** allows only string values today, **neither files nor attachment components**. Do you see any way to achieve this or a workaround?

![POST form-data with file](https://user-images.githubusercontent.com/13772138/181822749-79980ebd-61ac-4125-b029-ee8df3394b23.jpg)

**Reply by avivkri** [reply]
@melohagan,
Just wanted to check if you got around to the above question.
